docs/reference/aggregations/bucket/range-field-note.asciidoc

```diff
@@ -0,0 +1,173 @@
+PUT range_index
+{
+  "settings": {
+    "number_of_shards": 2
+  },
+  "mappings": {
+    "properties": {
+      "expected_attendees": {
+        "type": "integer_range"
+      },
+      "time_frame": {
+        "type": "date_range",
+        "format": "yyyy-MM-dd HH:mm:ss||yyyy-MM-dd||epoch_millis"
+      }
+    }
+  }
+}
+
+PUT range_index/_doc/1?refresh
+{
+  "expected_attendees" : {
+    "gte" : 10,
+    "lte" : 20
+  },
+  "time_frame" : {
+    "gte" : "2019-10-28 12:00:00",
+    "lte" : "2019-11-04"
+  }
+}
+POST /range_index/_search?size=0
+{
+    "aggs" : {
+        "range_histo" : {
+            "histogram" : {
+                "field" : "expected_attendees",
+                "interval" : 5
+            }
+        }
+    }
+}
+--------------------------------------------------
+
+Since the interval is 5 (and the offset is 0 by default), we expect buckets 10,
+15, and 20. Our range document will fall in all three of these buckets,
+something like this:
+
+[source, console-result]
+--------------------------------------------------
+{
+  ...
+  "aggregations" : {
+    "range_histo" : {
+      "buckets" : [
+        {
+          "key" : 10.0,
+          "doc_count" : 1
+        },
+        {
+          "key" : 15.0,
+          "doc_count" : 1
+        },
+        {
+          "key" : 20.0,
+          "doc_count" : 1
+        }
+      ]
+    }
+  }
+}
+--------------------------------------------------
+
+At present, there is no way for a document to exist partially in a bucket - we
+cannot count "one-third" in each of the above three buckets, and if we did a
+sub-aggregation to sum a value over these buckets, the full value would be
+included in each bucket.
+
+==== Query bounds are not aggregation filters
+
+A second unexpected behavior can arise when a query is used to filter on the
+field being aggregated.  In this case, a document could match the query, but
+still have one or both of the endpoints of the range outside the query.
+Consider the following aggregation on the above document:
+
+[source, console]
+--------------------------------------------------
+POST /range_index/_search?size=0
+{
+    "query": {
+      "range": {
+        "time_frame": {
+          "gte": "2019-11-01",
+          "format": "yyyy-MM-dd"
+        }
+      }
+    }, 
+    "aggs" : {
+        "november_data" : {
+            "date_histogram" : {
+                "field" : "time_frame",
+                "calendar_interval" : "day"
+              }
+        }
+    }
+}
+--------------------------------------------------
+
+Even though the query only considers days in November, the aggregation
+generates 8 buckets (4 in October, 4 in November), because the aggregation is
+calculated over the ranges of all matching documents:
+
+[source, console-result]
+--------------------------------------------------
+{
+  ...
+  "aggregations" : {
+    "november_data" : {
+      "buckets" : [
+        {
+          "key" : 1572220800000,
```

Comment:
Maybe toggle on the formatter so we have a `key_as_string` for easier viewing?

Reply:
I'd probably add annotations to the "unexpected" buckets too, to make it clearer.

You can do annotations with `<1>`, `<2>` inline with the JSON.

Then at the bottom of the snippet you can do:

```
--------------------------------------------------
<1> Text for annotation one
<2> Text for annotation two
```

You can also tag multiple things with the same annotation (e.g. multiple lines can be tagged with `<1>` so they are all annotated with the same item)

Reply:
Found a bug around the format parameter, see #47323, so we won't be doing that for this release.  I'll update the doc when I fix the bug.